presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/RowColumnValidator.java

```diff
@@ -0,0 +1,114 @@
+import static java.util.stream.Collectors.joining;
+
+public class RowColumnValidator
+        implements ColumnValidator
+{
+    @Inject
+    public RowColumnValidator()
+    {
+    }
+
+    @Override
+    public List<SingleColumn> generateChecksumColumns(Column column)
+    {
+        checkColumnType(column);
+
+        Expression checksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column.getIdentifier()));
+
+        ImmutableList.Builder<SingleColumn> columnsBuilder = ImmutableList.builder();
+        columnsBuilder.add(new SingleColumn(checksum, Optional.of(delimitedIdentifier(getChecksumAlias(column)))));
+
+        for (RowType.Field field : getFields(column)) {
+            Expression fieldChecksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(
+                    new DereferenceExpression(column.getIdentifier(), new Identifier(field.getName().get()))));
+            columnsBuilder.add(new SingleColumn(fieldChecksum, Optional.of(delimitedIdentifier(getChecksumAlias(column, field)))));
+        }
+
+        return columnsBuilder.build();
+    }
+
+    @Override
+    public ColumnMatchResult validate(Column column, ChecksumResult controlResult, ChecksumResult testResult)
+    {
+        checkColumnType(column);
+
+        ImmutableList<String> aliases = ImmutableList.<String>builder()
+                .add(getChecksumAlias(column))
+                .addAll(getFields(column).stream()
+                        .map(field -> getChecksumAlias(column, field))
+                        .collect(Collectors.toList()))
+                .build();
+
+        return new ColumnMatchResult(
+                aliases.stream().allMatch(alias -> Objects.equals(controlResult.getChecksum(alias), testResult.getChecksum(alias))),
+                prepareMatchMessage(aliases, controlResult, testResult));
+    }
+
+    private static void checkColumnType(Column column)
+    {
+        checkArgument(column.getType() instanceof RowType, "Expect RowType, found %s", column.getType().getDisplayName());
+    }
+
+    private static List<RowType.Field> getFields(Column column)
+    {
+        return ((RowType) column.getType()).getFields();
+    }
+
+    private static String getChecksumAlias(Column column)
+    {
+        return column.getName() + "_checksum";
+    }
+
+    private static String getChecksumAlias(Column column, RowType.Field field)
+    {
+        return column.getName() + "_" + field.getName().get() + "_checksum";
+    }
+
+    private static String prepareMatchMessage(ImmutableList<String> aliases, ChecksumResult controlResult, ChecksumResult testResult)
+    {
+        String control = aliases.stream()
+                .map(alias -> alias + ": " + controlResult.getChecksum(alias))
+                .collect(joining(", "));
+
+        String test = aliases.stream()
+                .map(alias -> alias + ": " + testResult.getChecksum(alias))
+                .collect(joining(", "));
```

Comment:
In case the RowType is a large structure, this would be too verbose to read. Let's only print the checksum where there is a mismatch.